File: Bymed.Application/Inventory/InventoryDto.cs
namespace Bymed.Application.Inventory;

public sealed record InventoryDto
{
    public required Guid ProductId { get; init; }
    public required string ProductName { get; init; }
    public string? Sku { get; init; }
    public int InventoryCount { get; init; }
    public int LowStockThreshold { get; init; }
    public bool IsAvailable { get; init; }
    public bool IsLowStock => InventoryCount <= LowStockThreshold;
}

File: bymed-admin/src/app/shared/models/inventory.dto.ts
export interface AdjustInventoryRequestDto {
  readonly adjustment: number;
  readonly reason: string;
}

/** Mirrors API `InventoryDto` (camelCase JSON). */
export interface InventoryItemDto {
  readonly productId: string;
  readonly productName: string;
  readonly sku?: string | null;
  readonly inventoryCount: number;
  readonly lowStockThreshold: number;
  readonly isAvailable: boolean;
  readonly isLowStock: boolean;
}
